src/core/agent/services/agentService.ts

```diff
@@ -1,11 +1,15 @@
-import { agentModules } from "../agent";
 import { Agent } from "@aries-framework/core";
+import { agentModules } from "../agent";
+import { BasicStoragesApi } from "../../storage/storage.types";
 
 abstract class AgentService {
   protected readonly agent: Agent<typeof agentModules>;
 
-  constructor(agent: Agent) {
+  protected readonly basicStorage!: BasicStoragesApi;
+
+  constructor(agent: Agent, basicStorage: BasicStoragesApi) {
     this.agent = agent;
+    this.basicStorage = basicStorage!;
```

Comment:
`genericRecords` is using with AriesAgent.agent.genericRecords... this format is okay?

Reply:
This format is OK yes but lets rename it to `get storage()` instead of `genericRecords`.

We will also rename it from `AriesAgent` but lets do that in a following PR as there will be a lot of UI refactors